tslearn/utils/cast.py

```diff
@@ -166,7 +169,7 @@ def to_seglearn_dataset(X):
     (10, 2)
     """
     X_ = check_dataset(X)
-    return numpy.array([Xi[:ts_size(Xi)] for Xi in X_])
+    return numpy.array([Xi[: ts_size(Xi)] for Xi in X_], dtype=object)
```

Comment:
Same

Reply:
Here I think that adding `dtype=object` is appropriate, the function `to_seglearn_dataset` gives exactly the expected result on the example proposed in the docstring.

Reply:
In fact this depends on how seglearn datasets should be formatted in case of series that are not all the same size. Could you check that ?

Reply:
We can find information in `seglearn` documentation:
https://dmbee.github.io/seglearn/user_guide.html#valid-sequence-data-representations
where it is written:
"Time series data can be represented as a list or array of arrays" with examples of arrays of different shapes.

This information is also confirmed by `tslearn`'s documentation the docstring of the function `to_seglearn_dataset`:
https://tslearn.readthedocs.io/en/stable/gen_modules/utils_conv/tslearn.utils.to_seglearn_dataset.html#tslearn.utils.to_seglearn_dataset
and in the docstring of the function `from_seglearn_dataset`:
https://tslearn.readthedocs.io/en/stable/gen_modules/utils_conv/tslearn.utils.from_seglearn_dataset.html#tslearn.utils.from_seglearn_dataset

@rtavenar, do you agree to use `dtype=object` here then?
It looks fine to me.